File: convert_data.py
import csv
import json
import os

csv_file_path = r"d:\Thực tập stuff\DF with React Only\Data Mart Kết quả học tập.csv"
output_file_path = r"d:\Thực tập stuff\DF with React Only\src\data.ts"

data = []

try:
    with open(csv_file_path, mode='r', encoding='utf-8') as csvfile:
        # The CSV seems to have headers based on the view_file output:
        # "gender","race_ethnicity","parental_education","math","math_score","reading","reading_score","writing","writing_score"
        reader = csv.DictReader(csvfile)
        
        for index, row in enumerate(reader):
            record = {
                "id": index + 1,
                "gender": row["gender"],
                "race_ethnicity": row["race_ethnicity"],
                "parental_education": row["parental_education"],
                "math_label": row["math"],
                "math_score": int(row["math_score"]),
                "reading_label": row["reading"],
                "reading_score": int(row["reading_score"]),
                "writing_label": row["writing"],
                "writing_score": int(row["writing_score"]),
                "status": "active",
                "lastUpdate": "2024-01-15"
            }
            data.append(record)

    ts_content = f"import {{ DataRecord }} from './App';\n\nexport const fullMockData: DataRecord[] = {json.dumps(data, indent=4)};"
    
    # We need to export the interface in App.tsx or redefine it here. 
    # To avoid circular dependency or type errors if I import from App, I'll just omit the type annotation in the export 
    # or ANY type, but better to just export as const and let inference work or use 'any'.
    # Actually, let's just make it a simple export.
    ts_content = f"export const fullMockData = {json.dumps(data, indent=4)};"

    with open(output_file_path, mode='w', encoding='utf-8') as outfile:
        outfile.write(ts_content)
    
    print(f"Successfully converted {len(data)} records to {output_file_path}")

except Exception as e:
    print(f"Error: {e}")

File: convert_data.js
const fs = require('fs');
const path = require('path');

const csvFilePath = "d:\\Thực tập stuff\\DF with React Only\\Data Mart Kết quả học tập.csv";
const outputFilePath = "d:\\Thực tập stuff\\DF with React Only\\src\\data.ts";

try {
    const csvContent = fs.readFileSync(csvFilePath, 'utf-8');
    const lines = csvContent.split(/\r?\n/).filter(line => line.trim() !== '');

    // Headers are: "gender","race_ethnicity","parental_education","math","math_score","reading","reading_score","writing","writing_score"
    // We skip the first line
    const data = [];

    // Regex to handle quoted strings properly: split by comma, but ignore commas inside quotes
    const splitCSV = (str) => {
        const result = [];
        let current = '';
        let inQuote = false;
        for (let i = 0; i < str.length; i++) {
            const char = str[i];
            if (char === '"') {
                inQuote = !inQuote;
            } else if (char === ',' && !inQuote) {
                result.push(current);
                current = '';
            } else {
                current += char;
            }
        }
        result.push(current);
        return result.map(s => s.replace(/^"|"$/g, '')); // Remove surrounding quotes
    };

    for (let i = 1; i < lines.length; i++) {
        const columns = splitCSV(lines[i]);
        if (columns.length < 9) continue;

        const record = {
            id: i,
            gender: columns[0],
            race_ethnicity: columns[1],
            parental_education: columns[2],
            math_label: columns[3],
            math_score: parseInt(columns[4]) || 0,
            reading_label: columns[5],
            reading_score: parseInt(columns[6]) || 0,
            writing_label: columns[7],
            writing_score: parseInt(columns[8]) || 0,
            status: "active",
            lastUpdate: "2024-01-15"
        };
        data.push(record);
    }

    const tsContent = `export const fullMockData = ${JSON.stringify(data, null, 4)};`;

    fs.writeFileSync(outputFilePath, tsContent);
    console.log(`Successfully converted ${data.length} records to ${outputFilePath}`);

} catch (err) {
    console.error('Error:', err);
}
